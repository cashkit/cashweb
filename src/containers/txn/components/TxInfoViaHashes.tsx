import React from 'react';


export interface TxInfoViaHashesProps {
    hash: Uint8Array | string,
    blockHash: Uint8Array | string,
    onClickHash: (txHash: Uint8Array | string) => void,
}

export const TxInfoViaHashes = ({
    hash,
    blockHash,
    onClickHash, }
    : TxInfoViaHashesProps) => {
  return(
    <>
      <div className="tile is-ancestor">
      <div className="tile is-parent">
        <article className="tile is-child box has-text-left notification is-primary">
        <p className="is-size-4 has-text-weight-medium">Tx Hash</p>
        <div className="content">{hash}</div>
        </article>
      </div>
      <div className="tile is-parent">
        <article className="tile is-child box has-text-left  is-info">
        <p className="is-size-4 has-text-weight-medium">Block Hash</p>
        <a className="content" onClick={() => onClickHash(blockHash)}>{blockHash}</a>
        </article>
      </div>
      </div>
    </>
  )
}
          
          
  