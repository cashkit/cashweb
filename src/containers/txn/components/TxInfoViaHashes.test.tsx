import React from 'react';
import renderer from 'react-test-renderer';

import { TxInfoViaHashes, TxInfoViaHashesProps } from './TxInfoViaHashes';

const props: TxInfoViaHashesProps = {
    hash: "",
    blockHash: "",
    onClickHash: (txHash: Uint8Array | string) => {},
}

it('<TxInfoViaHashes/> renders correctly', () => {
  const tree = renderer
    .create(<TxInfoViaHashes {...props}></TxInfoViaHashes>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});