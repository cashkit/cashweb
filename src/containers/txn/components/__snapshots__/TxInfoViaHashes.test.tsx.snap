// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TxInfoViaHashes/> renders correctly 1`] = `
<div
  className="tile is-ancestor"
>
  <div
    className="tile is-parent"
  >
    <article
      className="tile is-child box has-text-left notification is-primary"
    >
      <p
        className="is-size-4 has-text-weight-medium"
      >
        Tx Hash
      </p>
      <div
        className="content"
      >
        
      </div>
    </article>
  </div>
  <div
    className="tile is-parent"
  >
    <article
      className="tile is-child box has-text-left  is-info"
    >
      <p
        className="is-size-4 has-text-weight-medium"
      >
        Block Hash
      </p>
      <a
        className="content"
        onClick={[Function]}
      >
        
      </a>
    </article>
  </div>
</div>
`;
